src/Orleans.Core/Providers/LegacyProviderConfigurator.cs

```diff
@@ -0,0 +1,168 @@
+                    foreach (IProviderConfiguration providerConfig in providerGroup.SelectMany(kvp => kvp.Value.Providers.Values))
+                    {
+                        RegisterProvider<IStorageProvider>(providerConfig, services, defaultInitStage, defaultStartStage);
+                    }
+                }
+                else if (providerGroup.Key == ProviderCategoryConfiguration.LOG_CONSISTENCY_PROVIDER_CATEGORY_NAME)
+                {
+                    services.AddSingleton<ILogConsistencyProvider>(sp => sp.GetServiceByName<ILogConsistencyProvider>(ProviderConstants.DEFAULT_LOG_CONSISTENCY_PROVIDER_NAME));
+                    foreach (IProviderConfiguration providerConfig in providerGroup.SelectMany(kvp => kvp.Value.Providers.Values))
+                    {
+                        RegisterProvider<ILogConsistencyProvider>(providerConfig, services, defaultInitStage, defaultStartStage);
+                    }
+                }
+                else if (providerGroup.Key == ProviderCategoryConfiguration.STATISTICS_PROVIDER_CATEGORY_NAME)
+                {
+                    IProviderConfiguration providerConfig = providerGroup.SelectMany(kvp => kvp.Value.Providers.Values).FirstOrDefault();
+                    if(providerConfig != null)
+                    {
+                        // Looks like we only support a single statistics provider that can be any of the publisher interfaces. fml
+                        // TODO: Kill our statistics system.. please!?
+                        services.AddSingleton<IStatisticsPublisher>(sp => sp.GetServiceByName<IProvider>(providerConfig.Name) as IStatisticsPublisher);
+                        services.AddSingleton<ISiloMetricsDataPublisher>(sp => sp.GetServiceByName<IProvider>(providerConfig.Name) as ISiloMetricsDataPublisher);
+                        services.AddSingleton<IClientMetricsDataPublisher>(sp => sp.GetServiceByName<IProvider>(providerConfig.Name) as IClientMetricsDataPublisher);
+                        RegisterProvider<IProvider>(providerConfig, services, defaultInitStage, defaultStartStage);
+                    }
+                }
+                else
+                {
+                    foreach (IProviderConfiguration providerConfig in providerGroup.SelectMany(kvp => kvp.Value.Providers.Values))
+                    {
+                        RegisterProvider<IProvider>(providerConfig, services, defaultInitStage, defaultStartStage);
+                    }
+                }
+            }
+        }
+
+        private class ProviderLifecycleParticipant<TService> : ILifecycleParticipant<TLifeCycle>
+            where TService : class
+        {
+            private readonly ILogger logger;
+            private readonly IProviderConfiguration config;
+            private readonly IServiceProvider services;
+            private readonly LegacyProviderConfigurator.QueueTask schedualer;
```

Comment:
Since provider initialization logic in a silo needs be run in an Orleans scheduling context, but client does not, to avoid maintaining two classes that are mostly the same, I opted to make the scheduling plagable. and get it from the container.  Open to other suggestions.